### PR TITLE
Origin/adrian/frontend validation1

### PR DESCRIPTION
**Pull request by adrianchen91**
UX correction in next branch - e.g prompt to advise user which field is incorrect

### CODE REVIEW

File: client/modules/Registration/Registration.js [hunk 1]
```diff
@@ -3,33 +3,113 @@ import React, { Component, PropTypes } from 'react';
...
-                <label className={styles.labels} for="lastName">
-                    <div className={styles.labelText}>Last name *</div>
-                    <input type="text" name="lastName" id="lastName"/>
-                </label>
-                <label className={styles.labels} for="emailAddr">
+            <form onSubmit={this.handleSubmit}>
+                <label className={styles.labels} for="email">
                     <div className={styles.labelText}>Email *</div>
-                    <input type="text" name="emailAddr" id="emailAddr"/>
+                    <input className={errors.email ? "error" : ""} type="text" placeholder="Enter Email"
```
Comment (s3337852) [inline client/modules/Registration/Registration.js]: should `{errors.email ? "error" : ""}`
actually be `{errors.email ? style.error : ""}` ? 
i ran it locally and entered 'foo' and there was no update to the border of the input...

File: client/modules/Registration/Registration.js [hunk 2]
```diff
@@ -3,33 +3,113 @@ import React, { Component, PropTypes } from 'react';
...
+          password2: '',
+          mobile: '',
+          license: '',
+          
+          everFocusedEmail: false,
+          everFocusedPassword1: false,
+          everFocusedPassword2: false,
+          everFocusedMobile: false,
+          everFocusedLicense: false,
+          inFocus: '',
```
Comment (s3337852) [inline client/modules/Registration/Registration.js]: are we going to wire these up to something? i cant see them being used

File: client/modules/Registration/Registration.js [hunk 3]
```diff
@@ -3,33 +3,113 @@ import React, { Component, PropTypes } from 'react';
...
 
+var validator = require('validator');
+
 //Create a component class
+
+function validate(email, password1, password2, mobile, license) {
+    // true means invalid, so our conditions got reversed
+    return {
+      email: !validator.isEmail(email),
+      password1: password1.length === 0,
```
Comment (s3337852) [inline client/modules/Registration/Registration.js]: can we get a better password rule here? length > 0 is a bit relaxed :)

maybe use a regex test
this is an example for testing that it is at least 8 characters long, has an upper case, a lowercase and a number
"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$"

File: client/modules/Registration/Registration.js (hunk 1 above)
Comment (adrianchen91) [reply client/modules/Registration/Registration.js]: I couldn't get it to work either way for some reason

File: client/modules/Registration/Registration.js (hunk 2 above)
Comment (adrianchen91) [reply client/modules/Registration/Registration.js]: they were going to be used. but I will get rid of them in the interim

**Pull request closed without merging**